sympy/physics/mechanics/body.py

```diff
@@ -0,0 +1,168 @@
+__all__ = ['Body']
+
+
+class Body(RigidBody, Particle):
+    """
+    A Body which can be connected by joints.
+
+    Instances of Body can be connected using specific joints and can further
+    be used to generate equations of motion. It can be seen as an extension
+    of already present RigidBody and Particle classes and can inherit from
+    either of them based on the arguments passed.
+
+    Parameters
+    ----------
+    name: string
+        defines the name of the body. It is used as the base for defining body
+        specific properties.
+    masscenter : Point, optional
+        The point which represents the center of mass of the rigid body.
+    frame : ReferenceFrame (optional)
+        The ReferenceFrame which the rigid body is fixed in.
+    mass : Sympifyable, optional
+        The body's mass.
+    body_inertia : Dyadic (instance of inertia)
+        The body's inertia about center of mass.
+
+    Examples
+    --------
+    1. Default behaviour. It creates a RigidBody after defining mass,
+     masscenter, frame and inertia.
+
+    >>> from sympy.physics.mechanics import Body
+    >>> body = Body('name_of_body')
+
+    2. Passing attributes of Rigidbody. All the arguments needed to create a
+     RigidBody can be passed while creating a Body too.
+
+    >>> from sympy import Symbol
+    >>> from sympy.physics.mechanics import ReferenceFrame, Point, inertia
+    >>> from sympy.physics.mechanics import Body
+    >>> mass = Symbol('mass')
+    >>> masscenter = Point('masscenter')
+    >>> frame = ReferenceFrame('frame')
+    >>> body_inertia = inertia(frame, 1, 0, 0)
+    >>> body = Body('name_of_body', masscenter, mass, frame, body_inertia)
+
+    3. Creating a Particle. If masscenter and mass are passed, and inertia is
+     not then a Particle is created.
+
+    >>> from sympy import Symbol
+    >>> from sympy.physics.vector import Point
+    >>> from sympy.physics.mechanics import Body
+    >>> mass = Symbol('mass')
+    >>> masscenter = Point('masscenter')
+    >>> body = Body('name_of_body', masscenter, mass)
+
+    Similarly, A frame can also be passed while creating a Particle.
+
+    """
+    def __init__(self, name, masscenter=None, mass=None, frame=None,
+                 body_inertia=None):
+
+        self.name = name
+        self.force_list = []
+        self._counter = 0
+
+        if masscenter is None:
+            self._masscenter = Point(name + '_masscenter')
+        else:
+            self._masscenter = masscenter
+
+        if mass is None:
+            _mass = Symbol(name + '_mass')
+        else:
+            _mass = mass
+
+        if frame is None:
+            self._frame = ReferenceFrame(name + '_frame')
+        else:
+            self._frame = frame
+
+        if body_inertia is None and mass is None:
+            _inertia = (inertia(self._frame, 1, 1, 1), self._masscenter)
+        else:
+            _inertia = (body_inertia, self._masscenter)
+
+        self._masscenter.set_vel(self._frame, 0)
```

Comment:
I am adding masscenter in body's frame. I just want masscenter to be stationary in body's frame.